doc: Update 1.12.0 release guide and Changelog (#1488)
mongodb/terraform-provider-mongodbatlas: website/docs/guides/1.12.0-upgrade-guide.html.markdown

- You can now manage [Atlas Resource Tags](https://www.mongodb.com/docs/atlas/tags/)  in [`mongodbatlas_cluster`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cluster), [`mongodbatlas_advanced_cluster`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/advanced_cluster), and [`mongodbatlas_serverless_instance`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/serverless_instance). 
- You can now programmatically retrieve Atlas Organizational User details with [`mongodbatlas_atlas_user`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/data-sources/atlas_user) and [`mongodbatlas_atlas_users`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/data-sources/atlas_users) data sources 

> **Changes:**   

Reviewer: ```suggestion
**Breaking Changes:**   
```

> - Fix `mongodbatlas_database_user` to not require `roles.collection_name` attribute and not support empty `collection_name`. Users should remove any usage of `roles.collection_name = ""` in configuration for this resource for upgrading to this version.

Reviewer 2: similar changes as above
```suggestion
- Fixes `mongodbatlas_database_user` to not require `roles.collection_name` attribute and not support empty `collection_name`. You should remove any usage of `roles.collection_name = ""` in configurations for this resource when you upgrade to this version.
```
